db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryListOperatorNode.java

```diff
@@ -472,4 +473,24 @@ public int getOuterJoinLevel() {
     public void setOuterJoinLevel(int level) {
         outerJoinLevel = level;
     }
+
+    public ValueNode replaceIndexExpression(ResultColumnList childRCL) throws StandardException {
+        ValueNodeList newList = (ValueNodeList) getNodeFactory().getNode(
+                C_NodeTypes.VALUE_NODE_LIST,
+                getContextManager());
+        for (Object leftItem : leftOperandList) {
+            newList.addValueNode(((ValueNode) leftItem).replaceIndexExpression(childRCL));
+        }
+        leftOperandList = newList;
+        return this;
+    }
+
+    @Override
+    public boolean collectExpressions(Map<Integer, List<ValueNode>> exprMap) {
+        boolean result = true;
+        for (Object leftItem : leftOperandList) {
+            result = result && ((ValueNode) leftItem).collectExpressions(exprMap);
+        }
+        return result;
```

Comment:
We need to check the rightOperandList also, as it could also contain expressions:
For example:
```
create table t1 (a1 int, b1 int, c1 int);
create index idx_t1 on t1(a1+1);
create table t2(a2 int, b2 int, c2 int);

splice> explain select a1+1 from t1
> , t2 where a2 in (1, a1+1);
ERROR XJ001: DERBY SQL error: SQLCODE: -1, SQLSTATE: XJ001, SQLERRMC: java.lang.IllegalStateExceptionASSERT FAILED sourceResultSetNumber expected to be >= 0 for null.A1XJ001.U
```

Reply:
Done. The example is added as a test case.